### DISCUSSION TITLE
Help defining new mixture using python wrappers

### DISCUSSION THREAD

**Discussion by mina19** [usnistgov/REFPROP-issues#685] [Q&A] [answered]
Hello, I am not able to find any examples on defining a new mixture using REFPROP python wrappers.
If I have a fluid composition that is 98% ethane, 1.5 % methane, and 0.5% propane on a molar basis, how do I create that composition? 
I am trying to define this composition and then make calls to get its density, enthalpy, entropy given temperature and pressure. 

Code that I could copy/paste with explanatory comments would be extremely helpful as I'm a first-time REFPROP user. Thank you in advance.

**Comment by ianhbell** [comment, marked as answer]
Some info here: https://refprop-docs.readthedocs.io/en/latest/DLL/high_level.html#f/_/REFPROPdll .  There are also copious examples within the issues in this repository you should check out.

A snippet for your mixture to calculate the normal boiling point temperature of the saturated liquid state:
```python
import os
os.environ['RPPREFIX'] = os.getenv('HOME')+'/REFPROP10'

root = os.getenv('RPPREFIX')
import ctREFPROP.ctREFPROP as ct
RP = ct.REFPROPFunctionLibrary(root)
RP.SETPATHdll(root)

p = 101325
z = [0.98, 0.015, 0.005]
q = 0
r = RP.REFPROPdll('ETHANE;METHANE;PROPANE','PQ','T',RP.MOLAR_BASE_SI,0,0,p,q, z)
print('TNBP:', r.Output[0], r.hUnits)
```
yielding
```
TNBP: 174.91159925978994 K
```

**Reply by mina19** [reply]
Thank you for the quick response Ian!
I have a few questions... 

1. In the above example, what property is q? I see that we are passing a value of 0 for it and in the documentation it says that it means "quality".
2. If I want calculations performed with my user-defined units, what is the proper way to do that?

I tried 
p = 14.7
z = [0.98, 0.015, 0.005]
q = 0
r = RP.REFPROPdll('ETHANE;METHANE;PROPANE','PQ','T',RP.MOLAR_ENGLISH,0,0,p,q, z)
print('TNBP:', r.Output[0], r.hUnits)

and this returns the correct number in Fahrenheit.

But if I do:

USER = RP.GETENUMdll(0, "USER").iEnum

p = 0
z = [0.98, 0.015, 0.005]
q = 0
r = RP.REFPROPdll('ETHANE;METHANE;PROPANE','PQ','T',USER,0,0,p,q, z)
print('TNBP:', r.Output[0], r.hUnits)
this returns TNBP: -9999990.0 T which is incorrect.

where
                 USER (can be changed by calling the REFPROP subroutine)
iUnits --->      11
Temperature      C
Pressure         psig
Density          kg/m^3
Enthalpy         J/g
Entropy          (J/g)/K
Speed            m/s
Kinematic vis.   cm^2/s
Viscosity        mPa-s
Thermal cond.    W/(m-K)
Surface tension  N/m
Molar Mass       g/mol

**Comment by ianhbell** [comment]
I cannot recommend the use of USER units. Much better to use one of the base SI variants (molar or mass). It is fragile and has some bugs not present with the use of base-SI units. Also the use of base SI units avoids unit conversions in _your_ code.

q is the molar vapor quality, or the number of moles in the vapor phase relative to the total number of moles. It is 1 for a saturated vapor and 0 for a saturated liquid
